connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/ProtobufDeserializer.scala

```diff
@@ -157,6 +157,8 @@ private[sql] class ProtobufDeserializer(
 
       case (null, NullType) => (updater, ordinal, _) => updater.setNullAt(ordinal)
 
+      case (MESSAGE, NullType) => (updater, ordinal, _) => updater.setNullAt(ordinal)
```

Comment:
What is this for? For handling limited recursion?

Reply:
yes, correct.

Reply:
Could you add a comment about we might be dropping data here? It will not be easy to see for a future reader. 
We could have an option to error our if the actual data has more recursion than the configure.